### DISCUSSION TITLE
Trying to make error handling

### DISCUSSION THREAD

**Discussion by CustomEntity** [graphql-java/graphql-java#2850] [Q&A]
Hello, I would like to make it so that when a person makes a request to my function and the id does not match any entity, return an error. I didn't see anywhere in the documentation that you can do this. Is this possible?

```java    
private RuntimeWiring buildWiring() {
        return RuntimeWiring.newRuntimeWiring()
                .type(newTypeWiring("Query")
                        .dataFetcher("customers", environment ->
                                this.customerService.getCustomers())
                        .dataFetcher("customerById", environment -> {
                            // SEND ERROR IF NOT EXIST HERE
                        })
                ).build();
    }
```

**Comment by bbakerman** [comment]
You have a couple of choices but the one I recommend is `DataFetcherResult`

This allows you to return either data or an error (or technically both).  This special object is detected, unwrapped and the errors added to the list of returned errors.
